currency_rate_update_cmc/models/res_currency_rate_provider_CMC.py

```diff
@@ -0,0 +1,215 @@
+                    "BCA", "WA", "ANI", "XID", "JEW", "BT2", "VASH", "BIT",
+                    "TOKEN", "GMX", "GOD", "SPK", "VULC", "WOW", "CSC", "HC",
+                    "DAV", "ACES", "XRY", "SFE", "XQN", "EDRC", "ZBC", "FRN",
+                    "LEVO", "LEPEN", "SUP", "FONZ", "CYDER", "TCOIN", "CRYPT",
+                    "ROYAL", "DMC", "BAT", "SHA", "FLAP", "SLOTH", "WSX",
+                    "BIRDS", "BLAZR", "HIGH", "GRN", "GDC", "RBBT", "TOP",
+                    "RUNNERS", "ANTX", "MUSE", "UNRC", "ACN", "RICHX", "SKR",
+                    "PRN", "DON", "INDIA", "REGA", "XSTC", "KDC", "FID",
+                    "MARX", "XVE", "SJW", "LKC", "TURBO", "CMP", "TOPAZ",
+                    "APC", "GAIN", "ZSE", "LDCN", "NAMO", "PRIMU", "HNC",
+                    "NUMUS", "PAYP", "XTD", "BEST", "WINK", "AV", "SKULL",
+                    "BUB", "DCRE", "CME", "AKY", "UR", "GAY", "TELL", "HALLO",
+                    "FAZZ", "SIGMA", "SAK", "DUTCH", "SHELL", "MAGN", "OP",
+                    "HYPER", "BSN", "QBT", "CFC", "TODAY", "RUBIT", "POKE",
+                    "CHEAP", "DBG", "DASHS", "EGG", "OPES", "BET", "TEAM",
+                    "BAC", "NBIT", "MMXVI", "MONETA", "AXIOM", "MBL", "LTH",
+                    "FUTC", "BITOK", "BTBc", "SPORT", "CYC", "FRWC", "DUB",
+                    "GML", "CC", "VOYA", "QORA", "KARMA", "XAU", "TRICK",
+                    "IVZ", "X2", "PSY", "TCR", "DISK", "PRM", "HCC", "OMC",
+                    "LAZ", "RCN", "YES", "PDG", "KASHH", "RHFC", "TLE", "ASN",
+                    "UNC", "OCOW", "INF", "FRCT",
+                ]
+        return super()._get_supported_currencies(service)
+
+    allowed_base_currencies = [
+        "AUD", "BRL", "CAD", "CHF", "CLP", "CNY", "CZK", "DKK", "EUR", "GBP",
+        "HKD", "HUF", "IDR", "ILS", "INR", "JPY", "KRW", "MXN", "MYR", "NOK",
+        "NZD", "PHP", "PKR", "PLN", "RUB", "SEK", "SGD", "THB", "TRY", "TWD",
+        "USD", "ZAR"
+    ]
+
+    @api.model
+    def _obtain_rates(self, base_currency, currencies, date_from,
+                      date_to):
+        self.ensure_one()
+        if self.service != 'CMC':
+            return super()._obtain_rates(base_currency, currencies, date_from,
+                                         date_to)  # pragma: no cover
+        if base_currency not in self.allowed_base_currencies:
+            raise Exception('Company currency %s is not allowed in '
+                            'CoinMarketCap service ' % base_currency)
+        if base_currency in currencies:
+            currencies.remove(base_currency)
+
+        url = 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/' \
+              'quotes/historical'
+
+        content = defaultdict(dict)
+
+        parameters = {
+            'symbol': ','.join(currencies),
+            'time_start': date_from,
+            'time_end': date_to,
+            # 'count': time_end - time_start,
```

Comment:
This line is not needed?

Reply:
Well, I commented to remind me later to fix this. We need to add the `count` of days. If not, it will only get 10 days by default.